gammapy/maps/map_dict.py

```diff
@@ -0,0 +1,139 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from collections import UserDict
+from collections.abc import MutableMapping
+from astropy.io import fits
+from gammapy.maps import Map
+from gammapy.utils.scripts import make_path
+
+__all__ = ["Maps"]
+
+
+class Maps(MutableMapping):
+    """A Dictionary containing Map objects sharing the same geometry.
+
+    This class simplifies handling and I/O of maps collections.
+
+    For maps with different geometries, use a regular dict.
+    """
+
+    def __init__(self, **kwargs):
+        self._geom = None
+        self._data = {}
+        for key, value in kwargs.items():
+            self.__setitem__(key, value)
+
+    @property
+    def geom(self):
+        """Map geometry (`Geom`)"""
+        return self._geom
+
+    def __setitem__(self, key, value):
+        if value is not None and not isinstance(value, Map):
+            raise ValueError(f"MapDict can only contain Map objects.")
+
+        if len(self._data) > 0:
+            if value.geom != self._geom:
+                raise ValueError(f"MapDict items must share the same geometry.")
+        else:
+            self._geom = value.geom
+
+        self._data[key] = value
+
+    def __getitem__(self, key):
+        return self._data[key]
+
+    def __len__(self):
+        """Returns length of MapDict."""
+        return len(self._data)
+
+    def __delitem__(self, key):
+        del self._data[key]
+
+    def __iter__(self):
+        return iter(self._data)
+
+    def __repr__(self):
+        return f"{type(self).__name__}({self._data})"
+
```

Comment:
I think a `__str__` method with a little more output would be nice, like printing the geom and all the available maps...

Reply:
OK. Done.